File: src/main/java/cf/exception/BadRequestException.java
package pcrypto.cf.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

import java.util.List;


@ResponseStatus( HttpStatus.BAD_REQUEST )
public class BadRequestException
      extends AbstractApiException
{
    public BadRequestException( final String msg )
    {
        super( msg );
    }

    public BadRequestException( final String message,
                                final Throwable e )
    {
        super( message, e );
    }

    public BadRequestException( final String message,
                                final List<ErrorMessage> details )
    {
        super( message, details );
    }
}

File: src/main/kotlin/pcrypto/cf/exception/BadRequestException.kt
/*
 * Copyright (c) 2019 ChainFront LLC.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package pcrypto.cf.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus


@ResponseStatus(HttpStatus.BAD_REQUEST)
class BadRequestException : AbstractApiException {

    constructor(msg: String) : super(msg)

    constructor(
        message: String,
        e: Throwable
    ) : super(message, e)

    constructor(
        message: String,
        details: List<ErrorMessage>
    ) : super(message, details)

}
